lib/improver/metadata/set_attributes.py

```diff
@@ -136,7 +139,8 @@ def update_spot_title_attribute(cube):
                 "unable to replace grid description")
             return
         else:
-            if "UK" in original_title:
+            grid_descriptor = '{}'.format(title_regex.group('grid'))
```

Comment:
As far as I can tell for all the unit tests, the `grid_descriptor` is always an empty string, so the if statement `if "UK" in grid_descriptor` is never true. It would be good to have a test where this if statement is true.

Reply:
You're right - I got the regular expression wrong.  I've now fixed this and added tests to cover (including the unit test you suggest below).